.azure-devops/templates/set-testenv-sentinel.yml

```diff
@@ -15,7 +15,9 @@ steps:
             "azext_iot_testhub": os.environ.get("AZEXT_IOT_TESTHUB", sentinel_value),
             "azext_iot_testdps": os.environ.get("AZEXT_IOT_TESTDPS", sentinel_value),
             "azext_iot_teststorageuri": os.environ.get("AZEXT_IOT_TESTSTORAGEURI", sentinel_value),
-            "azext_iot_identity_teststorageid": os.environ.get("AZEXT_IOT_IDENTITY_TESTSTORAGEID", sentinel_value),
+            "azext_iot_identity_teststorageid": os.environ.get("AZEXT_IOT_IDENTITY_TESTSTORAGEID", sentinel_value),           
+            "azext_iot_teststorageconnstring": os.environ.get("AZEXT_IOT_TESTSTORAGECONNSTRING", sentinel_value),
```

Comment:
Can we look into auto-deriving the storage connection string? Looking to have as simple of test configuration as possible, with the least amount of user required info we can get by with.

Reply:
Auto deriving the connection string requires both storage account name as well as storage account key. Hence, instead of taking 2 inputs (storage account name and key) it is better to take only connection string as input.

The SAS URL can be generated using the connection string and container name, hence that input will no longer be needed

Reply:
> Auto deriving the connection string requires both storage account name as well as storage account key.

This is definitely not the case. It is common for resource providers to have/require "get keys" operations which can be used for show connection-string commands.

Storage supports this via the following command:

`az storage account show-connection-string` 

And it supports minimum identifiers...for example with just the storage account name.

```
user@machine:~$ az storage account show-connection-string --name mystorageaccount
{
  "connectionString": "DefaultEndpointsProtocol=https;EndpointSuffix=core.windows.net;AccountName=mystorageaccount;AccountKey=abcd12345"
}
```